src/types.ts

```diff
@@ -216,6 +219,10 @@ export type TransactionBody = {
     metadataHash?: FixlenBuffer<typeof METADATA_HASH_LENGTH>,
     validityIntervalStart?: Uint,
     mint?: Mint,
+    scriptDataHash?: FixlenBuffer<typeof SCRIPT_DATA_HAS_LENGTH>,
+    collaterals?: TransactionInput[],
```

Comment:
I suggest adding a `TransactionCollateralInput` type and using it here - we do so in Connect, so that we only have to modify `TransactionCollateralInput` instead of changing the type then (breaking change) in case IOHK adds another property to collateral inputs, or something like that.

Reply:
Ok, I don't know what will constitute a breaking change, so I'll rely on your knowledge here. Why is switching type different than changing the type?

Reply:
This is just my conjecture, but I think if an optional field is added to collateral inputs in the future, we can just add it to interop-lib's `TransactionCollateralInput` type and users of interop-lib don't have to do anything if they don't want to. However, changing the type later would force them to change the type in their code as well.

Reply:
I see your point. I'm too much of a native developer, thinking about ABIs, not in these higher level interfaces.